lib/arjdbc/common_jdbc_methods.rb

```diff
@@ -9,5 +9,19 @@ def initialize(connection, logger = nil, config = {})
 
       super(connection, logger, config)
     end
+
+    def translate_exception(e, message)
+      # we shall not translate native "Java" exceptions as they might
+      # swallow an ArJdbc / driver bug into a AR::StatementInvalid ...
+      return e if e.is_a?(NativeException) # JRuby 1.6
+      return e if e.is_a?(Java::JavaLang::Throwable)
+
+      case e
+        when ActiveModel::RangeError, SystemExit, SignalException, NoMemoryError then e
```

Comment:
is this seriously what AR is doing - they refactored but still ended up with mr. :hankey: ?

Reply:
Most of this is copied over from the existing jdbc adapter. I'm not sure how much of it is still necessary, but I figured it was best to make the sqlite adapter have the same base translations as the other adapters.

I believe I had to add the `ActiveModel::RangeError` check because we don't typecast the values until we are in the Java code, at which point any error gets passed through here. The `RangeError` was then being wrapped as a sql exception.

Reply:
understood, thanks for the explanation - thought you copied over from AR, if not its fine as is